pkg/pxc/app/configmap/configmap.go

```diff
@@ -0,0 +1,38 @@
+package configmap
+
+import (
+	"strings"
+
+	api "github.com/Percona-Lab/percona-xtradb-cluster-operator/pkg/apis/pxc/v1alpha1"
+	app "github.com/Percona-Lab/percona-xtradb-cluster-operator/pkg/pxc/app"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ConfigMap struct {
+	cm *corev1.ConfigMap
+}
+
+func NewConfigMap(cr *api.PerconaXtraDBCluster) *ConfigMap {
+	cm := &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "ConfigMap",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      strings.Join([]string{cr.Name, app.Name}, "-"),
+			Namespace: cr.Namespace,
+		},
+		Data: map[string]string{
+			"init.cnf": cr.Spec.PXC.Configuration,
+		},
+	}
+
+	return &ConfigMap{
+		cm: cm,
+	}
+}
+
+func (c *ConfigMap) ConfigMap() *corev1.ConfigMap {
+	return c.cm
+}
```

Comment:
I believe there is no point this additional complexity. For the time being it enough to return `corev1.ConfigMap` directly and use it in `ReconcilePerconaXtraDBCluster.deploy()`.